src/parsing/parser.rs

```diff
@@ -2598,7 +2598,18 @@ fn parse_jsx_opening_element<'a>(node: &'a JSXOpeningElement, context: &mut Cont
         if context.config.jsx_attributes_prefer_single_line {
             false
         } else if let Some(first_attrib) = node.attrs.first() {
-            node_helpers::get_use_new_lines_for_nodes(&node.name, first_attrib, context)
+            if first_attrib.leading_comments(context).count() != 0 {
+                return true
+            }
+            if context.config.jsx_attributes_prefer_hanging
+                && node_helpers::get_use_new_lines_for_nodes(&node.name, first_attrib, context) {
+                return true
+            }
+            if node.text(context).len() + first_attrib.text(context).len() <= context.config.line_width as usize {
```

Comment:
Ah, I see what you were asking now. This can't be done using `.text()` and `line_width` because someone could do something like this:

```ts
<Testing

    myAttribute={                                    test                         


                         }>
</Testing>
```

...which would require running the formatter twice in order to get it to format correctly.

So, inspecting the text as it currently is can't be done in many cases. It's done here because when `jsx_attributes_prefer_single_line` is false it allows the user to manually position how they want the attributes to show up.

Reply:
Oh, I got it. I never thought about code like it.
Thanks for the tip!